builtin/providers/aws/resource_aws_elasticache_replication_group.go

```diff
@@ -0,0 +1,306 @@
+	conn := meta.(*AWSClient).elasticacheconn
+
+	replicationGroupId := d.Get("replication_group_id").(string)
+	description := d.Get("description").(string)
+	cacheNodeType := d.Get("cache_node_type").(string)
+	automaticFailover := d.Get("automatic_failover").(bool)
+	numCacheClusters := d.Get("num_cache_clusters").(int)
+	engine := d.Get("engine").(string)
+	engineVersion := d.Get("engine_version").(string)
+	securityNameSet := d.Get("security_group_names").(*schema.Set)
+	securityIdSet := d.Get("security_group_ids").(*schema.Set)
+	subnetGroupName := d.Get("subnet_group_name").(string)
+	prefferedCacheClusterAZs := d.Get("preferred_cache_cluster_azs").(*schema.Set)
+
+	securityNames := expandStringList(securityNameSet.List())
+	securityIds := expandStringList(securityIdSet.List())
+	prefferedAZs := expandStringList(prefferedCacheClusterAZs.List())
+
+	req := &elasticache.CreateReplicationGroupInput{
+		ReplicationGroupID:          aws.String(replicationGroupId),
+		ReplicationGroupDescription: aws.String(description),
+		CacheNodeType:               aws.String(cacheNodeType),
+		AutomaticFailoverEnabled:    aws.Bool(automaticFailover),
+		NumCacheClusters:            aws.Int64(int64(numCacheClusters)),
+		Engine:                      aws.String(engine),
+		CacheSubnetGroupName:        aws.String(subnetGroupName),
+		EngineVersion:               aws.String(engineVersion),
+		CacheSecurityGroupNames:     securityNames,
+		SecurityGroupIDs:            securityIds,
+		PreferredCacheClusterAZs:    prefferedAZs,
+	}
+
+	if v, ok := d.GetOk("parameter_group_name"); ok {
+		req.CacheParameterGroupName = aws.String(v.(string))
+	}
+
+	_, err := conn.CreateReplicationGroup(req)
+	if err != nil {
+		return fmt.Errorf("Error creating Elasticache replication group: %s", err)
+	}
+
+	pending := []string{"creating"}
+	stateConf := &resource.StateChangeConf{
+		Pending:    pending,
+		Target:     "available",
+		Refresh:    ReplicationGroupStateRefreshFunc(conn, d.Id(), "available", pending),
+		Timeout:    60 * time.Minute,
+		Delay:      20 * time.Second,
+		MinTimeout: 5 * time.Second,
+	}
+
+	log.Printf("[DEBUG] Waiting for state to become available: %v", d.Id())
+	_, sterr := stateConf.WaitForState()
+	if sterr != nil {
+		return fmt.Errorf("Error waiting for elasticache (%s) to be created: %s", d.Id(), sterr)
+	}
+
+	d.SetId(replicationGroupId)
+
+	return nil
+}
+
+func resourceAwsElasticacheReplicationGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	req := &elasticache.DescribeReplicationGroupsInput{
+		ReplicationGroupID: aws.String(d.Id()),
+	}
+
+	res, err := conn.DescribeReplicationGroups(req)
+	if err != nil {
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "ReplicationGroupNotFoundFault" {
+			// Update state to indicate the replication group no longer exists.
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	if len(res.ReplicationGroups) == 1 {
```

Comment:
The ReplicationGroup status is in the output struct of the `DescribeReplicationGroups` method

- https://github.com/aws/aws-sdk-go/blob/master/service/elasticache/api.go#L4756

If I recall, the issue appeared when you reference `res.ReplicationGroups[0].NodeGroups[0].PrimaryEndpoint.Address`... but it could also be that some of the other elements simply weren't present. I was only able to trigger it by manually deleting the group in the console, then running `terraform refresh`